### PR TITLE
allow use of `ptls_get_cipher` during resumption

### PR DESCRIPTION
**Pull request by kazuho**
There is no reason to restrict access to the cipher-suite using which the client is trying to resume a session. And QUIC needs this.

So QUIC applications need to call `ptls_handshake` then call `ptls_get_cipher`. If a non-NULL value is returned, then resumption is in action. So the client may use the obtained cipher (and the exported key) to send 0-RTT data. If NULL is returned, then the session will never resume.

### CODE REVIEW

**Comment by kazuho** [comment]
@huitema Could you please review and merge if it works? Thank you in advance.

File: lib/picotls.c [hunk 1]
```diff
@@ -1156,12 +1154,14 @@ static int send_client_hello(ptls_t *tls, ptls_buffer_t *sendbuf, ptls_handshake
...
             uint32_t max_early_data_size;
-            if (decode_stored_session_ticket(tls->ctx, &resumption_cipher_suite, &resumption_secret, &obfuscated_ticket_age,
+            if (decode_stored_session_ticket(tls->ctx, &cipher_suite, &resumption_secret, &obfuscated_ticket_age,
                                              &resumption_ticket, &max_early_data_size, properties->client.session_ticket.base,
                                              properties->client.session_ticket.base + properties->client.session_ticket.len) == 0 &&
-                resumption_cipher_suite->hash == key_schedule_hash) {
+                cipher_suite->hash == key_schedule_hash) {
                 tls->client.offered_psk = 1;
+                tls->cipher_suite = cipher_suite;
                 if (max_early_data_size != 0 && properties->client.max_early_data_size != NULL) {
```
Comment (huitema) [inline lib/picotls.c]: Yes, that should do exactly what we want on the client side.

**Comment by huitema** [comment]
Verified that it works. I still have some work to do in picoquic, but I am doing a simple test, sending a PING frame as 0RTT data. I see on the logs that the frame is properly sent as 0RTT by the client, and is received and decrypted at the server. Still a bit of debugging to do, but it should be working tomorrow...
Thanks!

**Comment by huitema** [comment]
Picoquic client to other servers works with 0RTT -- tested with mozquic, nghttp2, winquic.

**Pull request merged**